lib/backend/k8s/resources/globalbgppeer.go

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"reflect"
+	"strings"
+
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/backend/k8s/thirdparty"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/converter"
+	"github.com/projectcalico/libcalico-go/lib/net"
+	"github.com/projectcalico/libcalico-go/lib/scope"
+
+	log "github.com/Sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	GlobalBGPPeerResourceName = "globalbgppeers"
+	GlobalBGPPeerTPRName      = "global-bgp-peer.projectcalico.org"
+)
+
+func NewGlobalBGPPeerHandler(c *kubernetes.Clientset, r *rest.RESTClient) ResourceHandler {
+	return newTPRHandler(
+		c, r,
+		GlobalBGPPeerTPRName,
+		GlobalBGPPeerResourceName,
+		"Calico Global BGP Peers",
+		reflect.TypeOf(thirdparty.GlobalBGPPeer{}),
+		reflect.TypeOf(thirdparty.GlobalBGPPeerList{}),
+		GlobalBGPPeerConverter{},
+	)
+}
+
+// GlobalBGPPeerConvert implements the ResourceConverter interface.
+type GlobalBGPPeerConverter struct {
+	converter.BGPPeerConverter
+}
+
+func (_ GlobalBGPPeerConverter) ListInterfaceToKey(l model.ListInterface) model.Key {
+	pl := l.(model.GlobalBGPPeerListOptions)
+	if pl.PeerIP.IP != nil {
+		return model.GlobalBGPPeerKey{PeerIP: pl.PeerIP}
+	}
+	return nil
+}
+
+func (_ GlobalBGPPeerConverter) KeyToResourceName(k model.Key) (string, error) {
+	// The name should be policed before we get here.
```

Comment:
Ok, so maybe it's trying to say "This method expects that the name has been validated by the caller", but this method doesn't take a name, it takes a Key and _returns_ a name, right?

So is it really "This method expects that the provided Key has been validated by the caller"?